wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/devfile/PreferencesDeserializerTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.workspace.server.devfile;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import java.io.Serializable;
+import java.util.Map;
+import org.mockito.Mock;
+import org.mockito.testng.MockitoTestNGListener;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+@Listeners(MockitoTestNGListener.class)
+public class PreferencesDeserializerTest {
+
+  @Mock private DeserializationContext ctxt;
+
+  private final JsonFactory factory = new JsonFactory();
+
+  /** Instance to test. */
+  private PreferencesDeserializer preferencesDeserializer = new PreferencesDeserializer();
+
+  @Test
+  public void shouldParseSimpleTypesCorrectly() throws Exception {
+    String json =
+        "{"
+            + "\"valid.string\": \"/usr/bin/value\","
+            + "\"valid.boolean\": false,"
+            + "\"valid.numeric\": 777555888"
```

Comment:
If the json schema is sticking with `number` types, we should add a floating point test case here.

Reply:
makes sense